src/control/createCollection.ts

```diff
@@ -0,0 +1,63 @@
+import { IndexOperationsApi } from '../pinecone-generated-ts-fetch';
+import type { ResponseError } from '../pinecone-generated-ts-fetch';
+import { mapHttpStatusError } from '../errors';
+import { builOptionConfigValidator } from '../validator';
+import { validIndexMessage } from './utils';
+
+import { Static, Type } from '@sinclair/typebox';
+
+const nonemptyString = Type.String({ minLength: 1 });
+
+const CreateCollectionOptionsSchema = Type.Object({
+  name: nonemptyString,
+  source: nonemptyString,
+});
+
+export type CreateCollectionOptions = Static<
```

Comment:
Yeah, the line length formatting has honestly been a big pain because `// @ts-ignore` doesn't work if things break across two lines and a lot of my test cases are specifically passing params that violate the typescript expectations to test method validation. Thinking about disabling that formatting rule.